### PR TITLE
refactor: $meta.saveAs

### PR DESCRIPTION
**Pull request by ranjith-sg**
Here,

1. the code which is used to save the resultset into a file by using stream option, is separated.
2. add csv output file, the csv $meta.saveAs is shown below
```
   $meta.saveAs = {
           filename: 'customers.csv',
           resultSetName: 'customers',
           columns: [{
                name: 'firstName',
                displayName: 'First Name',
                transform: (value, rowData) => { return value; }
           }]
    }
```
Now it support json and csv type output file and for other type of export, please extend this functionality by inheriting the Transform (saveAs/transform/index.js) class .

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -889,7 +839,8 @@ module.exports = function({parent}) {
                 }
             });
             if ($meta.saveAs) {
-                let fileDir = path.dirname($meta.saveAs);
+                var filename = typeof $meta.saveAs === 'string' ? $meta.saveAs : $meta.saveAs.filename;
+                let fileDir = path.dirname(filename);
```
Comment (kalinkrustev) [inline index.js]: We need to make sure that arbitrary paths cannot be used, due to security concerns and also isolation. This means saveAs should be a path relative to the working dir of the module (even better a subfolder), which is `path.join(this.bus.config.workDir, 'ut-port-sql', 'export')`

Here you can see how similar checks are done:
https://github.com/softwaregroup-bg/ut-port-file/blob/master/index.js#L62-L69

Comment (ranjith-sg) [reply index.js]: Okay I will made the required changes and will update here.

Comment (ranjith-sg) [reply index.js]: @kalinkrustev Can we keep the filename as simple name with ext, not as path ? ex. filename.csv. Becuase If the path is relative path ex(folder1/folder2/..../filename.ext), we need to handle to write dir(folder) by depthly. 

If it is simple filename, we just write 2 directory If not exists and next start to write the file stream.

Comment (kalinkrustev) [reply index.js]: I did not understand what is the issue with relative path. You can use fs-plus to create nested directories. Let's not limit the functionality. I think we may need to use sub-folders at some point.

Comment (ranjith-sg) [reply index.js]: I have made the required changes. please check it

**Comment by kalinkrustev** [comment]
@ranjith-sg please check lint errors
code review is complete, so I can merge once errors are fixed

**Comment by ranjith-sg** [comment]
@kalinkrustev , Please have a look at the follwiing lint error.

**Pull request merged**